File: Frontend/TodoFrontend/src/types/notification.ts
export interface Notification {
  id: number;
  title: string;
  message: string;
  task_id: number | null;
  user_id: number;
  read: boolean;
  sent: boolean;
  created_at: string;
  scheduled_for: string | null;
}

export interface NotificationSettings {
  id: number;
  user_id: number;
  email_notifications: boolean;
  task_reminders: boolean;
  reminder_time: number;
  daily_digest: boolean;
}

export interface NotificationSettingsUpdate {
  email_notifications?: boolean;
  task_reminders?: boolean;
  reminder_time?: number;
  daily_digest?: boolean;
}

export interface NotificationState {
  notifications: Notification[];
  unreadCount: number;
  settings: NotificationSettings | null;
  loading: boolean;
  error: string | null;
}

File: Backend-MS3/app/schemas/notification.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime

class NotificationBase(BaseModel):
    title: str
    message: str
    task_id: Optional[int] = None

class NotificationCreate(NotificationBase):
    user_id: int
    scheduled_for: Optional[datetime] = None

class NotificationUpdate(BaseModel):
    read: Optional[bool] = None
    sent: Optional[bool] = None

class NotificationResponse(NotificationBase):
    id: int
    user_id: int
    read: bool
    sent: bool
    created_at: datetime
    scheduled_for: Optional[datetime] = None

    class Config:
        from_attributes = True

class NotificationSettings(BaseModel):
    email_notifications: bool = True
    task_reminders: bool = True
    reminder_time: int = 24  # Hours before deadline
    daily_digest: bool = False

class NotificationSettingsResponse(NotificationSettings):
    id: int
    user_id: int

    class Config:
        from_attributes = True
